open_feature/hooks/hook.py

```diff
@@ -0,0 +1,66 @@
+from abc import abstractmethod
+
+from open_feature.evaluation_context.evaluation_context import EvaluationContext
+from open_feature.flag_evaluation.flag_evaluation_details import FlagEvaluationDetails
+from open_feature.flag_evaluation.flag_type import FlagType
+from open_feature.hooks.hook_context import HookContext
+
+
+class Hook:
+    @abstractmethod
+    def before(self, hook_context: HookContext, hints: dict) -> EvaluationContext:
+        """
+        Runs before flag is resolved.
+
+        :param hook_context: Information about the particular flag evaluation
+        :param hints: An immutable mapping of data for users to
+        communicate to the hooks.
+        :return: An optional EvaluationContext. If returned, it will
```

Comment:
This seems to indicate that the typehint above should be `typing.Optional[EvaluationContext]`

Reply:
Yeah I originally wrote it as an optional but it seemed unessecary, I'll change the comment